### PR TITLE
Docker e2e test fixes

### PR DESCRIPTION
**Pull request by gavin-aguiar**
## Description

Fixes #

---

### PR information

- [ ] The title of the PR is clear and informative.
- [ ] There are a small number of commits, each of which has an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [ ] If applicable, the PR references the bug/issue that it fixes in the description.
- [ ] New Unit tests were added for the changes made and CI is passing.

### Quality of Code and Contribution Guidelines
- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-functions-python-worker/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: .github/workflows/ci_docker_ded_workflow.yml [hunk 1]
```diff
@@ -5,6 +5,10 @@ name: CI Docker Dedicated tests
 
 on:
   workflow_dispatch:
+  push:
+    branches: [ dev, master, main, release/* ]
+  pull_request:
+    branches: [ dev, master, main, release/* ]
```
Comment (gavin-aguiar) [inline .github/workflows/ci_docker_ded_workflow.yml:11]: To be removed before merge

**Comment by gavin-aguiar** [comment]
run-docker-tests

File: .github/workflows/ci_consumption_workflow.yml
Comment (vrdmr) [inline .github/workflows/ci_consumption_workflow.yml:1]: Can we add loadscope as well? Such that -n contains the loadfile thing as well added? I understand this only has one file so is a no-op but if in the future there are multiple classes, we are safe.

Also, `-vv` for verbosity.

File: .github/workflows/ci_docker_con_workflow.yml
Comment (vrdmr) [inline .github/workflows/ci_docker_con_workflow.yml:1]: You can remove the code coverage stuff from the docker tests - as they won't be used or reported.

File: .github/workflows/ci_docker_ded_workflow.yml
Comment (vrdmr) [inline .github/workflows/ci_docker_ded_workflow.yml:1]: Similar comments as above.

File: codecov.yml [hunk 2]
```diff
@@ -1,3 +1,3 @@
 ignore:
-  - "tests/utils/testutils.py"
-  - "tests/utils/testutils_lc.py"
+  - "tests/utils/"
+
```
Comment (vrdmr) [inline codecov.yml:3]: For the future, can we explore [commenting](https://docs.codecov.com/docs/codecovyml-reference#comment)?

File: tests/endtoend/blob_functions/blob_functions_stein/function_app.py [hunk 3]
```diff
@@ -8,7 +8,7 @@
 
 import azure.functions as func
 
-app = func.FunctionApp()
+app = func.FunctionApp(http_auth_level=func.AuthLevel.ANONYMOUS)
```
Comment (vrdmr) [inline tests/endtoend/blob_functions/blob_functions_stein/function_app.py:11]: As discussed - moving `http_auth_level` to `auth_level` within `@app.route`? - All the places where you are adding `auth_level`

File: tests/endtoend/test_blob_functions.py [hunk 4]
```diff
@@ -131,22 +128,18 @@ def test_blob_trigger_with_large_content(self):
         # We check it every 2 seconds to allow the trigger to be fired
         max_retries = 10
         for try_no in range(max_retries):
-            time.sleep(2)
-
             try:
                 # Check that the trigger has fired
                 r = self.webhost.request('GET', 'get_blob_triggered')
+                time.sleep(2)
```
Comment (vrdmr) [inline tests/endtoend/test_blob_functions.py:134]: Add a comment explaining why "2" was added.

File: tests/endtoend/test_durable_functions.py [hunk 5]
```diff
@@ -1,9 +1,33 @@
...
+from tests.utils.constants import DEDICATED_DOCKER_TEST, \
+    CONSUMPTION_DOCKER_TEST
 
 
+@skipIf(is_envvar_true(DEDICATED_DOCKER_TEST)
+        or is_envvar_true(CONSUMPTION_DOCKER_TEST),
+        "Docker container cannot set webhook required "
+        "for durable tests")
 class TestDurableFunctions(testutils.WebHostTestCase):
+    env_variables = {}
```
Comment (vrdmr) [inline tests/endtoend/test_durable_functions.py:16]: Do we need to instantiate env_variables - if we can declare it within `WebHostTestCase` and use it using methods?

File: tests/endtoend/test_third_party_http_functions.py [hunk 6]
```diff
@@ -35,22 +35,21 @@ def setUpClass(cls):
             host_json = cls.get_script_dir() / 'host.json'
             with open(host_json, 'w+') as f:
                 f.write(HOST_JSON_TEMPLATE)
-            os_environ = os.environ.copy()
-            # Turn on feature flag
-            os_environ['AzureWebJobsFeatureFlags'] = 'EnableWorkerIndexing'
```
Comment (vrdmr) [inline tests/endtoend/test_third_party_http_functions.py:40]: Where are we setting `EnableWorkerIndexing` now?

File: .github/workflows/ci_consumption_workflow.yml
Comment (gavin-aguiar) [reply .github/workflows/ci_consumption_workflow.yml:1]: Done

File: .github/workflows/ci_docker_con_workflow.yml
Comment (gavin-aguiar) [reply .github/workflows/ci_docker_con_workflow.yml:1]: Done

File: .github/workflows/ci_docker_ded_workflow.yml
Comment (gavin-aguiar) [reply .github/workflows/ci_docker_ded_workflow.yml:1]: Done

File: tests/endtoend/test_third_party_http_functions.py (hunk 6 above)
Comment (gavin-aguiar) [reply tests/endtoend/test_third_party_http_functions.py:40]: In testutils and testutils_docker.py

File: tests/endtoend/test_durable_functions.py (hunk 5 above)
Comment (gavin-aguiar) [reply tests/endtoend/test_durable_functions.py:16]: Fixed

File: tests/endtoend/test_blob_functions.py (hunk 4 above)
Comment (gavin-aguiar) [reply tests/endtoend/test_blob_functions.py:135]: Done

File: tests/endtoend/test_dependency_isolation_functions.py [hunk 7]
```diff
@@ -32,12 +33,15 @@ class TestGRPCandProtobufDependencyIsolationOnDedicated(
...
         # Turn on feature flag
-        os_environ['PYTHON_ISOLATE_WORKER_DEPENDENCIES'] = '1'
+        cls.env_variables['PYTHON_ISOLATE_WORKER_DEPENDENCIES'] = '1'
+
         # Emulate Python worker in Azure environment.
-        # For how the PYTHONPATH is set in Azure, check prodV3/worker.py.
-        os_environ['PYTHONPATH'] = str(cls.customer_deps)
+        # For how the PYTHONPATH is set in Azure, check prodV4/worker.py.
+        cls.env_variables['PYTHONPATH'] = str(cls.customer_deps)
+
```
Comment (vrdmr) [inline tests/endtoend/test_dependency_isolation_functions.py:42]: Ideally shouldn't this portion be in the super's setupclass?

File: tests/endtoend/test_durable_functions.py [hunk 8]
```diff
@@ -1,10 +1,32 @@
...
+
+from azure_functions_worker.utils.common import is_envvar_true
 from tests.utils import testutils
+from tests.utils.constants import DEDICATED_DOCKER_TEST, CONSUMPTION_DOCKER_TEST
 
 
+@skipIf(is_envvar_true(DEDICATED_DOCKER_TEST)
+        or is_envvar_true(CONSUMPTION_DOCKER_TEST),
+        "This will be fixed in "
+        "https://github.com/Azure/azure-functions-python-worker/pull/1199")
```
Comment (vrdmr) [inline tests/endtoend/test_durable_functions.py:13]: Add a # TODO comment above such that it gets flagged in the IDE as well.

File: tests/endtoend/test_dependency_isolation_functions.py (hunk 7 above)
Comment (vrdmr) [reply tests/endtoend/test_dependency_isolation_functions.py:42]: Or this is a special case?

File: tests/endtoend/test_eventhub_batch_functions.py [hunk 9]
```diff
@@ -53,6 +66,7 @@ def test_eventhub_multiple(self):
 
         try:
             r = self.webhost.request('GET', 'get_eventhub_batch_triggered')
+            time.sleep(2)
```
Comment (vrdmr) [inline tests/endtoend/test_eventhub_batch_functions.py:69]: Comment next to it - explaining why sleep

File: tests/endtoend/test_dependency_isolation_functions.py (hunk 7 above)
Comment (gavin-aguiar) [reply tests/endtoend/test_dependency_isolation_functions.py:42]: No, this is specifically for this test case. All dependencies are in this dir dependency_isolation_functions

**Review by vrdmr — APPROVED** [review]

File: tests/endtoend/test_dependency_isolation_functions.py [hunk 10]
```diff
@@ -108,6 +121,9 @@ def test_paths_resolution(self):
             ).lower()
         )
 
+    @skipIf(is_envvar_true(DEDICATED_DOCKER_TEST)
+            or is_envvar_true(CONSUMPTION_DOCKER_TEST),
+            'Docker tests do not work with dependency isolation ')
```
[the remark this answers is not in the document]
Comment (gavin-aguiar) [reply tests/endtoend/test_dependency_isolation_functions.py]: Fixed

File: tests/endtoend/blob_functions/blob_functions_stein/function_app.py (hunk 3 above)
Comment (gavin-aguiar) [reply tests/endtoend/blob_functions/blob_functions_stein/function_app.py:11]: Will fix in the next PR

**Pull request merged**